Sources/PostgresNIO/New/PostgresQuery.swift

```diff
@@ -88,6 +88,26 @@ extension PostgresQuery {
             self.sql.append(contentsOf: "$\(self.binds.count)")
         }
 
+
+        @usableFromInline
+        struct InterpolatedCollectionCannotBeEmpty: Error {
+            @usableFromInline init() { }
+        }
+
+        @inlinable
+        public mutating func appendInterpolation<C: Collection>(
+            collection values: C
+        ) throws where C.Element: PostgresArrayEncodable {
+            guard !values.isEmpty else {
+                throw InterpolatedCollectionCannotBeEmpty()
+            }
+            let bindsSQL = try values.map { value in
+                try self.binds.append(value, context: .default)
+                return "$\(self.binds.count)"
+            }.joined(separator: ", ")
+            self.sql.append(contentsOf: "(\(bindsSQL))")
```

Comment:
This looks wrong to me... Why do we need to `for` loop this ourselves?

Does this sound better?

```suggestion
            try self.binds.append(value, context: .default)
            self.sql.append(contentsOf: "$\(self.binds.count)")
```

The problem currently is that it always go through the json path.

Reply:
> This looks wrong to me

Why?!

> Why do we need to for loop this ourselves?

That's the point, to produce something like 

```
SELECT my_column FROM my_table WHERE my_column in ($1, $2, $3, $4)
```

> Does this sound better?

Sound-wise yes, but i don't think it'll work with a _collection_. That's what the PR is trying to resolve.

> The problem currently is that it always go through the json path.

What goes through the JSON path? 🤔

Reply:
A query like
```sql
SELECT my_column FROM my_table WHERE my_column in $1
```
won't work. At least i couldn't get something like that to work.